components/powersensor/client.go

```diff
@@ -0,0 +1,97 @@
+package powersensor
+
+import (
+	"context"
+
+	"github.com/edaniels/golog"
+	pb "go.viam.com/api/component/powersensor/v1"
+	"go.viam.com/utils/rpc"
+	"google.golang.org/protobuf/types/known/structpb"
+
+	"go.viam.com/rdk/protoutils"
+	"go.viam.com/rdk/resource"
+)
+
+// client implements PowerSensorServiceClient.
+type client struct {
+	resource.Named
+	resource.TriviallyReconfigurable
+	resource.TriviallyCloseable
+	name   string
+	client pb.PowerSensorServiceClient
+	logger golog.Logger
+}
+
+// NewClientFromConn constructs a new Client from connection passed in.
+func NewClientFromConn(
+	ctx context.Context,
+	conn rpc.ClientConn,
+	remoteName string,
+	name resource.Name,
+	logger golog.Logger,
+) (PowerSensor, error) {
+	c := pb.NewPowerSensorServiceClient(conn)
+	return &client{
+		Named:  name.PrependRemote(remoteName).AsNamed(),
+		name:   name.ShortName(),
+		client: c,
+		logger: logger,
+	}, nil
+}
+
+func (c *client) Voltage(ctx context.Context, extra map[string]interface{}) (float64, bool, error) {
```

Comment:
This looks like a public function. I'm surprised the linter didn't yell at you for not having a comment describing it! Mild preference for putting one in explaining the return values; I had to read to the bottom of the function to realize that the boolean is not whether we successfully read the voltage, but rather whether the power is AC or DC.

Reply:
will add a comment above this and the above methods!